pkgs/tools/filesystems/sshfs-fuse/default.nix

```diff
@@ -1,52 +1,30 @@
-{ lib, stdenv, fetchFromGitHub
-, meson, pkg-config, ninja, docutils, makeWrapper
-, fuse3, glib
-, which, python3Packages
-, openssh
-}:
+{ lib, stdenv, callPackage, fetchpatch, darwin }:
 
-stdenv.mkDerivation rec {
-  version = "3.7.1";
-  pname = "sshfs-fuse";
+let mkSSHFS = args: callPackage (import ./common.nix args) {
+  inherit (darwin.apple_sdk.frameworks) DiskArbitration;
+};
+in if stdenv.isDarwin then
+  mkSSHFS {
+    version = "2.10"; # macFUSE isn't yet compatible with libfuse 3.x
+    sha256 = "1dmw4kx6vyawcywiv8drrajnam0m29mxfswcp4209qafzx3mjlp1";
+    patches = [
+      # remove reference to fuse_darwin.h which doens't exist on recent macFUSE
+      ./fix-fuse-darwin-h.patch
```

Comment:
Can we apply this patch uncondionaly to prevent bitrot?

Reply:
I separated the commits because sshfs on Darwin is currently stuck on 2.10, which is almost 4 years old. I thought it'd be better if we attempt to have common patches after macFUSE gains support for libfuse 3.x.

Ideally, this patch should be upstreamed, but I haven't created a PR because the latest sshfs isn't going to build against macFUSE anyways.

Reply:
I didn't even see that the version is different.